### PR TITLE
ColorManagement: Rename .legacyMode=false → .enabled=true

### PR DESCRIPTION
**Pull request by donmccurdy**
Related issue:

- https://github.com/mrdoob/three.js/pull/24700

Renames `ColorManagement.legacyMode = false` to `ColorManagement.enabled = true`. While I think it's fair to say that three.js "is color managed" in some sense with or without this flag, the .legacyMode terminology is confusing. I'd frame this as a flag that enables/disables the THREE.ColorManagement API, specifically, rather than enabling/disabling all colorspace-aware decisions globally (which no one wants).

Also updates the `webgl_mirror.html` example to use `ColorManagement.enabled = true`. Because hexadecimal and CSS-like colors are assumed to be sRGB in that mode (as they are in HTML/CSS) it is necessary to update the currently-specified colors, doing either of two things:

- (A) Keep using hexadecimal syntax; convert Linear-sRGB → sRGB
- (B) Specify colors as RGB components; no conversion needed

I've opted for (A) here.

### CODE REVIEW

**Comment by LeviPesin** [comment]
I oppose this (and the previous) PR for two reasons:
1) Setting `ColorManagement.enabled = false` feels like disabling the color management in three.js completely, which is not the case - only hexadecimal, fog, background, and clear colors management is changed. `legacyMode` suits more, because it means that we had some color management mode before, and that we have another now.
2) `ColorManagement.enabled = true` feels like enabling some optional mode, not that one that will become default at some time. If we want more users to migrate to the new mode, it would be better to note that the old mode is legacy even in the property name. (Something like `disableLegacy` can also work)
Also (not very related to this PR, but coming from the idea in 2), maybe we can log a deprecation warning if ColorManagement is used in the legacy mode?

File: src/math/ColorManagement.js
Comment (Mugen87) [inline src/math/ColorManagement.js:27]: Please add the release when the property has been renamed so we can easily removed the code after ten releases.
```js
console.warn( 'THREE.ColorManagement: .legacyMode=false has been renamed to .enabled=true. in r147' );
```

Comment (Mugen87) [reply src/math/ColorManagement.js:27]: Example:

https://github.com/mrdoob/three.js/blob/735d376fc58a3b78d58fdb8dacd47bd6fd4b2d69/src/core/BufferAttribute.js#L346-L352

**Comment by donmccurdy** [comment]
Thanks @LeviPesin — 

> maybe we can log a deprecation warning if ColorManagement is used in the legacy mode?

Do you mean logging a warning for all apps that do not specify `ColorManagement.enabled = true` (or equivalent?)? It's probably too soon for that, we haven't even started using it in examples... Or only warn if `ColorManagement` is accessed in some way? The current ColorManagement methods are mostly used internally, so I'm not sure what problem this would solve yet, or how to detect internal vs. external access.

> Setting ColorManagement.enabled = false feels like disabling the color management in three.js completely, which is not the case...

I can't argue about what it feels like — that's certainly going to depend on the reader, and I understand your reaction. I guess I'd just point out that others have found the `legacyMode` naming to be confusing in other ways. We may have to just choose a name that is "good enough", then document what it does and does not affect.

`THREE.ColorManagement.enabled = true` matches the convention of `THREE.Cache.enabled = true`, for example, and that flag certainly does not toggle all cache-like behavior in the library. The [THREE.Cache](https://threejs.org/docs/?q=cache#api/en/loaders/Cache) system describes its behavior, and the `.enabled` flag enables/disables that behavior.

**Comment by LeviPesin** [comment]
> Do you mean logging a warning for all apps that do not specify ColorManagement.enabled = true (or equivalent?)?

Yes — but if it is too soon currently, maybe in a few releases?

**Comment by donmccurdy** [comment]
No need to convince me, I like the API already. :) 

Practically I think we'll want to make the change in some number of examples, and see what issues or feedback comes up, either from users or maintainers. 

Assuming we decide to move forward, the other question is whether to make the change in isolation, or to time it alongside any other changes. Changing these two defaults at the same time would reduce friction, compared to changing either one alone::


- `THREE.ColorManagement.enabled = true`
- `renderer.outputEncoding = sRGBEncoding`<sup>1</sup>

<sup>1</sup> or similarly, `renderer.outputColorSpace = SRGBColorSpace`, with https://github.com/mrdoob/three.js/pull/23936

**Review by Mugen87 — APPROVED** [review]

**Comment by donmccurdy** [comment]
Rebased and updated the Editor — I'll merge the PR this week, if that's OK.

**Comment by Mugen87** [comment]
Would be nice to see this change in `R150`!

**Pull request merged**
